Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -504,7 +505,7 @@ extension DatadogContextProvider {
         serverDateProvider: ServerDateProvider
     ) {
         let context = DatadogContext(
-            site: configuration.site,
+            site: configuration.site ?? .us1,
```

Comment:
suggestion/ Defaults resolution is error-prone and dangerous if done in different places. For that reason, in V1 convention was to resolve defaults early in configuration declaration - so we don't need to struggle with optionals in downstream logic.

I think the root cause of optional here is:
```swift
/// [Datadog Site](https://docs.datadoghq.com/getting_started/site/) for data uploads. It can be `nil` in V1
/// if the SDK is configured using deprecated APIs: `set(logsEndpoint:)`, `set(tracesEndpoint:)` and `set(rumEndpoint:)`.
let site: DatadogSite?
```
I'd recommend tackling this legacy early and individually (isolated PR would be best). If we remove these deprecated APIs on `v2` branch, it will streamline further changes around `site` logic.

Reply:
Good point, I agree 👍